nav2_smac_planner/include/nav2_smac_planner/node_lattice.hpp

```diff
@@ -86,15 +85,39 @@ struct LatticeMotionTable
   float num_angle_quantization_float;
   float min_turning_radius;
   float bin_size;
+  std::vector<float> primitive_start_angles;
   float change_penalty;
   float non_straight_penalty;
   float cost_penalty;
   float reverse_penalty;
   ompl::base::StateSpacePtr state_space;
   std::vector<TrigValues> trig_values;
   std::string current_lattice_filepath;
+  std::vector<std::vector<MotionPrimitive>> motionPrimitives;
+  LatticeMetadata latticeMetadata;
 };
 
+/**
+ * @brief convert json to lattice metadata 
+ * @param[in] j json object
+ * @param[out] lattice meta data 
+ */
+void fromJsonToMetaData(const nlohmann::json &j, LatticeMetadata &latticeMetaData);
```

Comment:
Do these 3 function need to be inside of this object to access local variables? If not, I'd recommend moving them to the utils.hpp file since they're kind of utility-esk

Reply:
Done, very clever use of inline. I though it was irrelevant because the compiler would handle _inlining_. It was pretty cool to learn that it would resolve functions that are defined twice within a translation unit.